scheduler/test/e2e/scheduler_test.go

```diff
@@ -36,118 +43,165 @@ var _ = Describe("[E2E] KeptnScheduler", Ordered, func() {
-
-			pod = WithContainer(st.MakePod().
-				Namespace("default").
-				Name(name).
-				Req(map[apiv1.ResourceName]string{apiv1.ResourceMemory: "50"}).
-				ZeroTerminationGracePeriod().
-				Obj(), pause)
-
-			err := k8sClient.Create(ctx, pod)
-			Expect(ignoreAlreadyExists(err)).NotTo(HaveOccurred(), "could not add pod")
+			*pod1 = initPod(*pod1, nil, labels1, KeptnScheduler)
+			*pod2 = initPod(*pod2, nil, labels2, KeptnScheduler)
 		})
 
 		Context("a new Pod", func() {
+			It(" should stay pending if no workload instance is available", func() {
+				checkPending(pod1)
+			})
+			It(" should be scheduled when workload instance pre-evaluation checks are done", func() {
+				checkWorkload("mylabeledapp-myworkload-1.0.1", *pod1, "Succeeded")
+			})
 
-			It(" should be immediately scheduled", func() {
-
-				Eventually(func() error {
-					return podRunning(pod.Namespace, pod.Name)
-				}).Should(Succeed())
-
+			It(" should NOT be scheduled when workload instance pre-evaluation checks fails", func() {
+				checkWorkload("mylabeledapp-myworkload-1.0.2", *pod2, "Failed")
 			})
 		})
 	})
+
 })
 
-func initWorkloadInstance() *testv1alpha2.KeptnWorkloadInstance {
+func initPod(pod apiv1.Pod, annotations map[string]string, labels map[string]string, scheduler string) apiv1.Pod {
+
+	//create a test Pod
+	name := names.SimpleNameGenerator.GenerateName("my-testpod-")
+	pause := imageutils.GetPauseImageName()
+	pod = *WithContainer(st.MakePod().
+		Namespace("default").
+		Name(name).
+		Req(map[apiv1.ResourceName]string{apiv1.ResourceMemory: "5"}).
+		ZeroTerminationGracePeriod().
+		Obj(), pause)
+	if annotations != nil {
+		pod.Annotations = annotations
+	}
+	if labels != nil {
+		pod.Labels = labels
+	}
+	if scheduler == KeptnScheduler {
+		pod.Spec.SchedulerName = scheduler
+	}
+
+	err := k8sClient.Create(ctx, &pod)
+	Expect(ignoreAlreadyExists(err)).NotTo(HaveOccurred(), "could not add pod")
+	return pod
+}
+
+func checkPending(pod *apiv1.Pod) {
+
+	newPod := &apiv1.Pod{}
+	Eventually(func() error {
+		err := k8sClient.Get(ctx, types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}, newPod)
+		return err
+	}).Should(Succeed())
+
+	Expect(newPod.Status.Phase).To(Equal(apiv1.PodPending))
+
+}
+
+func checkWorkload(workloadname string, pod apiv1.Pod, status common2.KeptnState) {
+	workloadinstance := initWorkloadInstance(workloadname)
+
+	err := k8sClient.Create(ctx, workloadinstance)
+	Expect(ignoreAlreadyExists(err)).To(BeNil())
+
+	Eventually(func() error {
+		err := k8sClient.Get(ctx, types.NamespacedName{Namespace: pod.Namespace, Name: workloadname}, workloadinstance)
+		return err
+	}).Should(Succeed())
+	workloadinstance.Status.PreDeploymentEvaluationStatus = status
+	err = k8sClient.Status().Update(ctx, workloadinstance)
+
+	Expect(err).To(BeNil())
+	assertion := Eventually(func() error {
+		return podScheduled(pod.Namespace, pod.Name)
+	}).WithTimeout(time.Second * 20).WithPolling(3 * time.Second)
```

Comment:
this looks like a common assertion, I am wondering if it makes sense to provide a function for this or not 🤔 
WDYT?